packages/vm/src/cache.rs

```diff
@@ -69,22 +77,26 @@ where
         })
     }
 
-    pub fn save_wasm(&mut self, wasm: &[u8]) -> VmResult<Vec<u8>> {
-        check_wasm(wasm)?;
-        let id = save(&self.wasm_path, wasm)?;
+    pub fn save_wasm(&mut self, wasm: &[u8]) -> VmResult<Checksum> {
+        check_wasm(wasm, &self.supported_features)?;
+        let checksum = save(&self.wasm_path, wasm)?;
         let module = compile(wasm)?;
-        let hash = WasmHash::generate(&id);
+        let module_hash = checksum.derive_module_hash();
         // singlepass cannot store a module, just make best effort
```

Comment:
this comment is obsolete (fixed in 0.7 I believe)

Reply:
Ah, good to know. I stumbled upon it a couple of time and did not understand. Will delete.

Reply:
Does this also mean we stop ignoring the error next line?

Reply:
Yes, this should work for all modules now. And replacing `let _ =` with a trailing `?` would be good.